### PR TITLE
Use Hazelcast for Events Tasks and sessions #7752

### PR DESCRIPTION
**Pull request by rymsha**

### CODE REVIEW

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/cluster/ClusterEventReceiver.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/NullSessionStoreFactoryActivator.java  1
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/HazelcastSessionStoreFactoryActivatorTest.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/WebSessionStoreConfigServiceImpl.java  4
- modules/web/web-jetty/src/main/java/com/enonic/xp/web/jetty/impl/session/JettySessionStoreConfiguratorImpl.java  1
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/NullSessionStoreFactoryActivatorTest.java  1
- modules/core/core-hazelcast/src/test/java/com/enonic/xp/core/impl/hazelcast/HazelcastActivatorTest.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/AllTasksReporter.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/SingleTaskReporter.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/ClusteredTaskManagerImpl.java  6
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/HazelcastConfigServiceImpl.java  4
- modules/core/core-hazelcast/src/test/java/com/enonic/xp/core/impl/hazelcast/HazelcastConfigServiceImplTest.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/RunningTasksReporter.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/OsgiSupport.java  7
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/WebSessionStoreConfigServiceImplTest.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/TasksReporterCallable.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/HazelcastActivator.java  2
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/HazelcastSessionStoreFactoryActivator.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/AbstractSessionStoreFactoryActivator.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5060559&bid=16655208)

File: modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/cluster/ClusterEventReceiver.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    private String registrationId;
+
+    @Activate
+    public ClusterEventReceiver( @Reference final HazelcastInstance hazelcastInstance, @Reference final EventPublisher eventPublisher )
+    {
+        this.topic = hazelcastInstance.getTopic( ClusterEventSender.ACTION );
+        this.eventPublisher = eventPublisher;
+    }
+
+    @Activate
```
Comment (anatol-sialitski) [inline modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/cluster/ClusterEventReceiver.java]: @rymsha why do you use two times `@Activate` annotation?

Comment (rymsha) [reply modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/cluster/ClusterEventReceiver.java]: One for "constructor injection", another for real activation logic.

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/AllTasksReporter.java  1
- modules/core/core-hazelcast/src/test/java/com/enonic/xp/core/impl/hazelcast/HazelcastConfigServiceImplTest.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/WebSessionStoreConfigServiceImpl.java  4
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/HazelcastSessionStoreFactoryActivator.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/ClusteredTaskManagerImpl.java  6
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/OsgiSupport.java  7
- modules/web/web-jetty/src/main/java/com/enonic/xp/web/jetty/impl/session/JettySessionStoreConfiguratorImpl.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/AbstractSessionStoreFactoryActivator.java  3
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/NullSessionStoreFactoryActivator.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/SingleTaskReporter.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/HazelcastConfigServiceImpl.java  4
- modules/core/core-hazelcast/src/test/java/com/enonic/xp/core/impl/hazelcast/HazelcastActivatorTest.java  1
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/cluster/ClusterEventReceiver.java  1
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/WebSessionStoreConfigServiceImplTest.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/HazelcastActivator.java  2
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/TasksReporterCallable.java  1
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/HazelcastSessionStoreFactoryActivatorTest.java  1
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/NullSessionStoreFactoryActivatorTest.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/RunningTasksReporter.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5060559&bid=16655208)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/WebSessionStoreConfigServiceImplTest.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/AllTasksReporter.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/NullSessionStoreFactoryActivator.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/RunningTasksReporter.java  1
- modules/core/core-hazelcast/src/test/java/com/enonic/xp/core/impl/hazelcast/HazelcastConfigServiceImplTest.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/HazelcastConfigServiceImpl.java  4
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/NullSessionStoreFactoryActivatorTest.java  1
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/cluster/ClusterEventReceiver.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/AbstractSessionStoreFactoryActivator.java  3
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/OsgiSupport.java  7
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/SingleTaskReporter.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/HazelcastActivator.java  2
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/ClusteredTaskManagerImpl.java  6
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/TasksReporterCallable.java  1
- modules/web/web-jetty/src/main/java/com/enonic/xp/web/jetty/impl/session/JettySessionStoreConfiguratorImpl.java  1
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/HazelcastSessionStoreFactoryActivatorTest.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/WebSessionStoreConfigServiceImpl.java  4
- modules/core/core-hazelcast/src/test/java/com/enonic/xp/core/impl/hazelcast/HazelcastActivatorTest.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/HazelcastSessionStoreFactoryActivator.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5060559&bid=16655208)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/HazelcastSessionStoreFactoryActivatorTest.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/HazelcastConfigServiceImpl.java  4
- modules/core/core-hazelcast/src/test/java/com/enonic/xp/core/impl/hazelcast/HazelcastConfigServiceImplTest.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/TasksReporterCallable.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/SingleTaskReporter.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/HazelcastActivator.java  2
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/NullSessionStoreFactoryActivator.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/OsgiSupport.java  7
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/AllTasksReporter.java  1
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/WebSessionStoreConfigServiceImplTest.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/HazelcastSessionStoreFactoryActivator.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/RunningTasksReporter.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/WebSessionStoreConfigServiceImpl.java  4
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/NullSessionStoreFactoryActivatorTest.java  1
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/cluster/ClusterEventReceiver.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/ClusteredTaskManagerImpl.java  6
- modules/web/web-jetty/src/main/java/com/enonic/xp/web/jetty/impl/session/JettySessionStoreConfiguratorImpl.java  1
- modules/core/core-hazelcast/src/test/java/com/enonic/xp/core/impl/hazelcast/HazelcastActivatorTest.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5060559&bid=16655208)

**Comment by jsi** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/SingleTaskReporter.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/AllTasksReporter.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/ClusteredTaskManagerImpl.java  6
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/HazelcastSessionStoreFactoryActivator.java  1
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/RunningTasksReporter.java  1
- modules/web/web-jetty/src/main/java/com/enonic/xp/web/jetty/impl/session/JettySessionStoreConfiguratorImpl.java  1
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/HazelcastSessionStoreFactoryActivatorTest.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/HazelcastConfigServiceImpl.java  4
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/WebSessionStoreConfigServiceImplTest.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/WebSessionStoreConfigServiceImpl.java  4
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/distributed/TasksReporterCallable.java  1
- modules/core/core-event/src/main/java/com/enonic/xp/core/impl/event/cluster/ClusterEventReceiver.java  1
- modules/core/core-hazelcast/src/main/java/com/enonic/xp/core/impl/hazelcast/HazelcastActivator.java  2
- modules/core/core-task/src/main/java/com/enonic/xp/impl/task/OsgiSupport.java  7
- modules/core/core-hazelcast/src/test/java/com/enonic/xp/core/impl/hazelcast/HazelcastConfigServiceImplTest.java  1
- modules/web/web-session/src/test/java/com/enonic/xp/web/session/impl/NullSessionStoreFactoryActivatorTest.java  1
- modules/core/core-hazelcast/src/test/java/com/enonic/xp/core/impl/hazelcast/HazelcastActivatorTest.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/NullSessionStoreFactoryActivator.java  1
- modules/web/web-session/src/main/java/com/enonic/xp/web/session/impl/AbstractSessionStoreFactoryActivator.java  3
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/enonic/xp/pullRequest?prid=5060559&bid=16655208)

**Pull request merged**